src/main/java/teammates/logic/core/FeedbackSessionsLogic.java

```diff
@@ -1164,6 +1164,31 @@ public void updateFeedbackSession(FeedbackSessionAttributes newSession)
 
         fsDb.updateFeedbackSession(newSession);
     }
+
+    /**
+     * Decrease the response rate of feedback session by 1 if the deleted response is the last response from
+     * the particular student in that feedback session; for use after deleting a student response
+     * @param studentEmail
+     * @param sessionName
+     * @param courseId
+     */
+    public void updateSessionResponseRateForDeletedStudentResponse(String studentEmail, String sessionName,
+            String courseId) throws InvalidParametersException, EntityDoesNotExistException {
+        if (!hasRemainingResponsesFromStudent(studentEmail, sessionName, courseId)) {
+            deleteStudentFromRespondentList(studentEmail, sessionName, courseId);
+        }
+    }
```

Comment:
Anyway, it feels like it should be a private method because it is step of a bigger operation and one that should only be called as part of that big operation.